File: SpringBootAOP/src/main/java/com/example/web/TestController.java
package com.example.web;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.service.TestService;

@RestController
public class TestController {

	@Autowired
	TestService service;
	
	@GetMapping(value = "/noAop")
	public String noAop(){
		return service.test();
	}
	
	@GetMapping(value = "/aop")
	public String aop(){
		return service.testAop();
	}
}

File: SpringBootAOP/src/main/kotlin/com/example/kotlin/web/TestController.kt
package com.example.kotlin.web

import com.example.kotlin.service.TestService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class TestController {

    @Autowired
    private lateinit var service: TestService

    @GetMapping(value = ["/noAop"])
    fun noAop(): String {
        return service.test()
    }

    @GetMapping(value = ["/aop"])
    fun aop(): String {
        return service.testAop()
    }
}
